test/async-emitter.js

```diff
@@ -0,0 +1,262 @@
+'use strict';
+
+const { AsyncEmitter } = require('..');
+const metatests = require('metatests');
+
+metatests.test('AsyncEmitter on/emit', async test => {
+  const ae = new AsyncEmitter();
+
+  const fn = test.mustCall(async (a, b, c, d) => {
+    test.strictSame(a, 1);
+    test.strictSame(b, 2);
+    test.strictSame(c, 3);
+    test.strictSame(d, 4);
+  });
+
+  ae.on('e1', fn);
+
+  await ae.emit('e1', 1, 2, 3, 4);
+
+  test.strictSame(ae.count('e1'), 1);
+  test.strictSame(ae.names().length, 1);
+
+  test.end();
+});
+
+metatests.test('AsyncEmitter once', async test => {
+  const ae = new AsyncEmitter();
+
```

Comment:
You should add a test, to ensure that `emit` args are forwarded to `once` listener.